engine/src/main/java/org/terasology/world/block/items/BlockItemFactory.java

```diff
@@ -85,37 +98,145 @@ public EntityRef newInstance(BlockFamily blockFamily, int quantity) {
         return builder.build();
     }
 
+    /**
+     * Create a new block item for the given {@link BlockFamily}, with {@code blockEntity} as reference entity to retain
+     * components from.
+     * <p>
+     * The item quantity defaults to 1.
+     * <p>
+     * Use {@link #newBuilder(BlockFamily, EntityRef, int)} if you want to modify the block item entity's properties
+     * before it gets created.
+     *
+     * @param blockFamily block family to create the block item builder for
+     * @param blockEntity reference block entity to retain components from
+     * @return the block item entity
+     */
     public EntityRef newInstance(BlockFamily blockFamily, EntityRef blockEntity) {
         if (blockFamily == null) {
             return EntityRef.NULL;
         }
 
+        return createBuilder(blockFamily, blockEntity, (byte) 1).build();
+    }
+
+    /**
+     * Create a new block item builder for the given {@link BlockFamily} and item quantity.
+     * <p>
+     * Attempts to resolve the corresponding block prefab to retrieve a list of potential components to add.
+     * <p>
+     * Use this method if you want to modify the block item entity's properties before it gets created.
+     *
+     * @param blockFamily block family to create the block item builder for
+     * @param quantity item quantity (see {@link ItemComponent#stackCount}); constrained to [0...128)
+     * @return a pre-populated entity builder for a block item entity
+     */
+    public EntityBuilder newBuilder(BlockFamily blockFamily, int quantity) {
+        final ComponentContainer components =
+                blockFamily.getArchetypeBlock().getPrefab()
+                        .map(p -> ((ComponentContainer) p))
+                        .orElse(EntityRef.NULL);
+
+        return createBuilder(blockFamily, components, (byte) Ints.constrainToRange(quantity, 0, Byte.MAX_VALUE));
```

Comment:
I feel like there should be a shorter form of that conversion. Oh, here we go:

https://guava.dev/releases/snapshot/api/docs/com/google/common/primitives/SignedBytes.html#saturatedCast-long-

`signedBytes.saturatedCast(quantity)`